### DISCUSSION TITLE
Notelock shake on classic mod does not match stable

### DISCUSSION THREAD

**Discussion by ghost** [ppy/osu#20223] [Ideas] [answered]
There is a small visual difference between notelocked hitobjects on stable and lazer. On stable, it'll shake a bit when you attempt to click it, however on lazer, no visual feedback is given.

See the attached comparison video. 

https://user-images.githubusercontent.com/102317848/189494243-938dbd2e-5c3b-45ef-99fa-c2536d46eca7.mp4

**Comment by smoogipoo** [comment, marked as answer]
Can you provide a beatmap + replay? It would be easier to understand what's going on if so.

**Reply by ghost** [reply]
(apologies for the late reply)

I actually lost the original replay file, so I had to remake the replay. Both the replay and the beatmap should be in the attached zip file.

If it was unclear, the problem is that the hitobjects on lazer aren't shaking at all, whereas they are on stable. 

[notelock.zip](https://github.com/ppy/osu/files/9570587/notelock.zip)

**Reply by peppy** [reply]
Thanks for that. The shake is being attempted, but lazer thinks it doesn't have enough time to perform the shake:

![JetBrains Rider 2022-09-21 at 06 31 59](https://user-images.githubusercontent.com/191335/191430982-00b48bf2-ee33-4541-b78b-e5df163739b6.png)

I'm not sure the [shake length limiting logic](https://github.com/ppy/osu/blob/f8830c6850128456266c82de83273204f8b74ac0/osu.Game.Rulesets.Osu/Objects/Drawables/DrawableHitCircle.cs#L142) makes sense, so I'll take a look into this one.

**Reply by peppy** [reply]
https://github.com/ppy/osu/pull/20398

**Comment by peppy** [comment]
I can't actually see the lazer version shaking at all. As above, a replay would be helpful.
